### DISCUSSION TITLE
[Feature] Introduce multi-tenancy

### DISCUSSION THREAD

**Discussion by GunniBusch** [Performave/convoy#100] [Feature Requests & Suggestions]
Hi,

Is there any plan or possibility to add multi-tenancy support?

Implementing this feature would allow us to have a central storage for nodes/IPs/locations , while being able to spawn separate instances or tenants that are based on the main configuration. This would be particularly useful for using the platform as a reselling interface, though considerations such as permissions would need to be addressed.

I would be willing to assist with this, although I don’t have the time to familiarize myself with the current database model structure. However, there are packages available that handle much of the functionality, such as routing (e.g., https://tenancyforlaravel.com).

**Comment by ericwang401** [comment]
I have interest in multi tenancy but I'm not pursuing it at the moment. 

There are packages to assist with the implementation like the one you mentioned but I still have to develop the front-end and the permission system.

Single database tenancy can make the database less normalized because I have some unique constraints on some columns like email for example. If a user signed up on one hosting provider, they can't ever be a customer on another provider.

So this just boils down to I have to do a lot of research on this bc I don't know much abt multitenancy, but I also don't have the availability to work on this either.
